icgcget/cli.py

```diff
@@ -71,6 +55,38 @@ def logger_setup(logfile):
     return logger
 
 
+def check_code(client, code):
+    if code != 0:
+        logger.error("{} client exited with a nonzero error code {}.".format(client, code))
+        raise click.ClickException("Please check client output for error messages")
```

Comment:
I tend to agree with you there. That's why we typically use a technique called "Exception translation", whereby a higher layer translates an exception from a lower layer. If you do this right, you should only need one such handler. The handler can then map to the appropriate response / action.